ros/ros_comm
# Changes between 1.15.8 and 1.15.9 for backporting to Melodic

The following list of changes has been integrated into ros_comm 1.15.9 (Noetic) since the last Melodic release (1.14.9)

**Backported** (in this PR):

* Add skip_cache parameter to rosnode_ping() (#2009)
  * Useful enhancement

* Remove unavailable nodes from local node cache _caller_apis (#2010)
  * Bug fix

* Update local parameter cache on set_param (#2021)
  * Bug fix

* Fix issue with rospy.set_param('') (#2024)
  * Bug fix

* Fix Lost Wake Bug in ROSOutAppender (#2033)
  * Bug fix

* Change error message (#2035)
  * Useful enhancement

* Fix log*_throttle with sim time (#2044)
  * Bug fix

* XmlRpcValue::_doubleFormat should be used during write (#2003)
  * Bug fix

* Gracefully stop recording upon SIGTERM and SIGINT (#2038)
  * Bug fix

* Install advertisetest (#2045)
  * Bug fix

* Fix spelling (#2066)
  * Documentation fixes

* Trap for overly large input to XmlRPCPP (#2065)
  * Bug fix

* Cached parameter should be unsubscribed (#2068)
  * Bug fix

* [Windows] Single quote string fix-up (#2051)
  * Bug fix

* Update maintainers (#2075)

* Fix paramtest for empty values of a parameter (#2054)
  * Bug fix

* Set call_finished_ with true for each call inside callFinished (#2074)
  * Bug fix

**Not backported:**

* Replace xrange with range (#2013)
  * Not necessary, Python 3

* Ignore underscores when parsing literal numeric values #2022
  * Not necessary, Python 3

* Use heapq.merge instead of custom merge sort code (#2017)
  * Python 3 only

* Update boost::placeholders usage for boost 1.73 (and later) (#2023)
  * Not necessary, for boost >= 1.73

* Fix error handling with Python 3 (#2050)
  * Not necessary, Python 3
